AltaPayApi/AltaPayApi/Service/Requests/Common/BasePaymentRequestRequest.cs

```diff
@@ -20,6 +20,13 @@ protected BasePaymentRequestRequest()
 		public string Language { get; set; }
 		public IDictionary<string,object> PaymentInfos { get; set; }
 		public AuthType Type { get; set; }
+		public AgreementType? AgreementType { get; set; }
+		public string AgreementId { get; set; }
+		public AgreementUnscheduledType? AgreementUnscheduledType { get; set; }
+		public string AgreementExpiry { get; set; }
+		public string AgreementFrequency { get; set; }
+		public string AgreementNextChargeDate { get; set; }
+		public string AgreementAdminUrl { get; set; }
```

Comment:
Why not using a complex object, like the PaymentRequestConfig

Reply:
Great idea, done, thank you.